lms/djangoapps/teams/static/teams/js/collections/topic.js

```diff
@@ -16,12 +15,13 @@
                 },
 
                 constructor: function(topics, options) {
-                    BaseCollection.prototype.constructor.call(this, topics, options);
-
-                    this.state.pageSize = topics.results.length;
                     if (topics.sort_order) {
                         this.state.sortKey = topics.sort_order;
                     }
+
+                    options.perPage = topics.results.length;
+                    BaseCollection.prototype.constructor.call(this, topics, options);
+
```

Comment:
This is why the calculation of totalPages was wrong, as we were first calling baseCollection constructor which took the default pageSize and we were overriding the pageSize after the initial models were passed to the collection.